common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java

```diff
@@ -190,6 +217,52 @@ public void onFailure(Throwable t) {
     }
   }
 
+  private <T> void retry(
+      final int retryCountValue,
+      final int saslRetryCountValue,
+      final int maxRetries,
+      final boolean enableSaslRetries,
+      int delayMs,
+      Supplier<CompletableFuture<T>> action,
+      CompletableFuture<T> future) {
+    action.get()
+            .thenAccept(future::complete)
+            .exceptionally(e -> {
+              int retryCount = retryCountValue;
+              int saslRetryCount = saslRetryCountValue;
+              boolean isIOException = e instanceof IOException
+                      || (e.getCause() != null && e.getCause() instanceof IOException);
+              boolean isSaslTimeout = enableSaslRetries && e instanceof SaslTimeoutException;
+              if (!isSaslTimeout && saslRetryCount > 0) {
+                Preconditions.checkState(retryCount >= saslRetryCount,
+                        "retryCount must be greater than or equal to saslRetryCount");
+                retryCount -= saslRetryCount;
```

Comment:
Could you add comment to explain the relationship between `retryCount ` and `saslRetryCount`? I don't quite get it.

Reply:
Add some comment.

Consistent with the retry logic of the RetryingBlockTransferor.shouldRetry method.

https://github.com/apache/spark/blob/80bba4463eba29a56cdd90642f0681c3710ce87c/common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RetryingBlockTransferor.java#L249-L253